docs/src/migration.md

```diff
@@ -0,0 +1,334 @@
+# MetaGraphs.jl to MetaGraphsNext.jl
+
+This section is mainly targeted towards people intending to migrate from
+*MetaGraphs.jl* to *MetaGraphsNext.jl*. We present how one could implement the
+examples shown in *MetaGraphs.jl*'s
+[Example Usage](https://juliagraphs.org/MetaGraphs.jl/dev/#Example-Usage)
+section using *MetaGraphsNext.jl*.
+
+***
+
+###### Import packages
+
+```@example main
+using Graphs
+import MetaGraphs, MetaGraphsNext
+```
+
+```@meta
+DocTestSetup = quote
+  using Graphs
+  import MetaGraphsNext
+end
+```
+
+***
+
+###### Create a metagraph based on a simplegraph, with optional default edge weight
+
+Using *MetaGraphs.jl*:
+```@example main
+_mg = MetaGraphs.MetaGraph(path_graph(5), 3.0)
+```
+
+Using *MetaGraphsNext.jl*:
+```jldoctest main
+mg = MetaGraphsNext.MetaGraph(Graph(), default_weight = 3.0)
+
+add_vertex!(mg, :a, nothing)
+add_vertex!(mg, :b, nothing)
+add_vertex!(mg, :c, nothing)
+add_vertex!(mg, :d, nothing)
+add_vertex!(mg, :e, nothing)
+
+add_edge!(mg, :a, :b, nothing)
+add_edge!(mg, :b, :c, nothing)
+add_edge!(mg, :c, :d, nothing)
+add_edge!(mg, :d, :e, nothing)
+mg
+
+# output
+
+Meta graph based on a {5, 4} undirected simple Int64 graph with vertex labels of type Symbol, vertex metadata of type Nothing, edge metadata of type Nothing, graph metadata given by nothing, and default weight 3.0
+```
+
+***
+
+###### Create a directed metagraph based on the simplegraph, with optional default edge weight
+
+Using *MetaGraphs.jl*:
+```@example main
+_mdg = MetaGraphs.MetaDiGraph(path_graph(5), 3.0)
+```
+
+Using *MetaGraphsNext.jl*:
+```jldoctest main
+mg = MetaGraphsNext.MetaGraph(DiGraph(), default_weight = 3.0)
+
+# A more concise way to construct a path graph with 5 vertices
+foreach(x -> add_vertex!(mg, Symbol(x), nothing), 'a':'e')
+foreach(x -> add_edge!(mg, Symbol(x[1]), Symbol(x[2]), nothing), zip('a':'e', 'b':'e'))
+mg
+
+# output
+
+Meta graph based on a {5, 4} directed simple Int64 graph with vertex labels of type Symbol, vertex metadata of type Nothing, edge metadata of type Nothing, graph metadata given by nothing, and default weight 3.0
+```
+
+***
+
+###### Set some properties for the graph itself
+
+Using *MetaGraphs.jl*:
+```@example main
+MetaGraphs.set_prop!(_mg, :description, "This is a metagraph.")
+```
+
+Using *MetaGraphsNext.jl*:
+```jldoctest main
+# Important: graph data can only be set when constructing the object
```

Comment:
This is arguably a bug. We could make just this one field mutable on more recent versions of Julia.